### PR TITLE
Adds + operator for `String?`

### LINKED ISSUE #28: Add `+` operator that works with string optionals
I tried this out in a playground and it seems to work well enough. This task is basically a reminder for me to add it and write some tests.

```swift
func +(lhs: String?, rhs: String?) -> String? {
    guard let lhs = lhs else { return rhs }
    guard let rhs = rhs else { return lhs }
    return lhs + rhs
}
```

### PR DESCRIPTION
**Pull request by SerenadeX**
Per @nickmshelley's suggestion, this will add the `+` operator to `String?`s.  Fixes #28

### CODE REVIEW

File: SwiftificationTests/StringTests.swift [hunk 1]
```diff
@@ -87,4 +87,21 @@ class StringTests: XCTestCase {
...
     }
     
-}
\ No newline at end of file
+    func testAppendAndPrependOptionalString() {
+        let str1: String? = "Hi"
+        let str2: String? = " Bob "
+        let strNil: String? = nil
+        
+        let prependString1On2 = str1 + str2 // "Hi Bob "
```
Comment (nickmshelley) [inline SwiftificationTests/StringTests.swift]: Naming these doesn't really buy you anything because you can just do `XCTAssertEqual(str1 + str2, "Hi Bob ")`, which is both shorter and (IMO) clearer.

File: SwiftificationTests/StringTests.swift [hunk 2]
```diff
@@ -87,4 +87,21 @@ class StringTests: XCTestCase {
...
+        let appendString1On2 = str2 + str1 // " Bob Hi"
+        let onlyString1 = str1 + strNil // "Hi"
+        let onlyString1Append = strNil + str1 // "Hi"
+        
+        XCTAssertEqual(prependString1On2, "Hi Bob ")
+        XCTAssertEqual(appendString1On2, " Bob Hi")
+        XCTAssertEqual(onlyString1, "Hi")
+        XCTAssertEqual(onlyString1Append, "Hi")
```
Comment (nickmshelley) [inline SwiftificationTests/StringTests.swift]: We should also test with two `nil` values. And I think it would be useful to test with an optional string on one side and a non-optional on the other (both combinations), just in case that adds some type weirdness.

**Comment by nickmshelley** [comment]
.

File: Swiftification/String.swift [hunk 3]
```diff
@@ -52,5 +52,11 @@ public extension String {
         }
         return result
     }
-    
-}
\ No newline at end of file
+
+}
+
+public func +(lhs: String?, rhs: String?) -> String? {
```
Comment (hiltonc) [inline Swiftification/String.swift]: Add a space after +. This convention aids in readability for custom operators.

File: SwiftificationTests/StringTests.swift [hunk 4]
```diff
@@ -87,4 +87,21 @@ class StringTests: XCTestCase {
...
+        let nonNilString = "I'm not Nil!"
+        XCTAssertEqual(str1 + str2, "Hi Bob ")
+        XCTAssertEqual(str2 + str1, " Bob Hi")
+        XCTAssertEqual(str1 + strNil, "Hi")
+        XCTAssertEqual(strNil + str1, "Hi")
+        XCTAssertEqual(nonNilString + str1, "I'm not Nil!Hi")
+        XCTAssertEqual(str1 + nonNilString, "HiI'm not Nil!")
+        XCTAssertEqual(strNil + nonNilString, "I'm not Nil!")
+        XCTAssertEqual(nonNilString + strNil, "I'm not Nil!")
+
```
Comment (hiltonc) [inline SwiftificationTests/StringTests.swift]: Remove these two empty lines, as well as the one at the end of the class.
